numpy/lib/function_base.py

```diff
@@ -770,29 +770,68 @@ def select(condlist, choicelist, default=0):
     array([ 0,  1,  2,  0,  0,  0, 36, 49, 64, 81])
 
     """
-    n = len(condlist)
-    n2 = len(choicelist)
-    if n2 != n:
+    # Check the size of condlist and choicelist are the same, or abort.
+    if len(condlist) != len(choicelist):
         raise ValueError(
-            "list of cases must be same length as list of conditions")
-    choicelist = [default] + choicelist
-    S = 0
-    pfac = 1
-    for k in range(1, n+1):
-        S += k * pfac * asarray(condlist[k-1])
-        if k < n:
-            pfac *= (1-asarray(condlist[k-1]))
-    # handle special case of a 1-element condition but
-    #  a multi-element choice
-    if type(S) in ScalarType or max(asarray(S).shape) == 1:
-        pfac = asarray(1)
-        for k in range(n2+1):
-            pfac = pfac + asarray(choicelist[k])
-        if type(S) in ScalarType:
-            S = S*ones(asarray(pfac).shape, type(S))
-        else:
-            S = S*ones(asarray(pfac).shape, S.dtype)
-    return choose(S, tuple(choicelist))
+            'list of cases must be same length as list of conditions')
+
+    # Now that the dtype is known, handle the deprecated select([], []) case
+    if len(condlist) == 0:
+        warnings.warn("select with an empty condition list is not possible"
+                      "and will be deprecated",
+                      DeprecationWarning)
+        return np.asarray(default)[()]
```

Comment:
What's the rationale for deprecating this?

Reply:
Don't remember, maybe you would not know the result shape?

Reply:
Or the correct dtype, the comment above seems like an oversight that should be removed

Reply:
@eric-wieser 

What would be the semantic meaning of numpy.select without any basis for selecting?

by analogy:    1+    <--- what is the meaning of + without a thing to add?

Reply:
The problem is not the basis of selecting, but the shape of what is "inside" choicelist being unknown. I think this was actually about choicelist being empty more then condlist being empty probably.

To be honest, it would somewhat make sense to allow it if default is passed in and can be used to get the shape. That seems dubious though (and even if, it seems like this needs to be deprecated first).

Anyway, I am starting to have memories of a similar discussion, so nvm. if Eric is still curious will look at it again.

Reply:
> To be honest, it would somewhat make sense to allow it if default is passed in and can be used to get the shape. 

This is the point I was making. By having a default, the operation becomes well-defined - in the same way that `max([])` is an error in python, but `max([], default=0)` is not

Reply:
> What would be the semantic meaning of numpy.select without any basis for selecting?

Phase it as _"choose `choicelist[i]` if `condlist[i]` is `True`. If not any of `condlist[i]` are `True`, choose `default"_, and the semantic meaning is clear, because `not any([])` is defined as True